File: TextosJapones/src/main/java/org/jisho/textosJapones/database/dao/ComicInfoDao.java
package org.jisho.textosJapones.database.dao;

import org.jisho.textosJapones.model.entities.comicinfo.ComicInfo;
import org.jisho.textosJapones.model.exceptions.ExcessaoBd;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

public interface ComicInfoDao {

    void insert(ComicInfo obj) throws ExcessaoBd;

    void update(ComicInfo obj) throws ExcessaoBd;

    ComicInfo select(String comic, String linguagem) throws ExcessaoBd;
    ComicInfo select(UUID id, String comic, String linguagem) throws ExcessaoBd;

    List<ComicInfo> selectEnvio(LocalDateTime ultimo) throws ExcessaoBd;

}

File: ProcessaTexto/src/main/kotlin/br/com/fenix/processatexto/database/dao/ComicInfoDao.kt
package br.com.fenix.processatexto.database.dao

import br.com.fenix.processatexto.model.entities.comicinfo.ComicInfo
import java.sql.SQLException
import java.time.LocalDateTime
import java.util.*

interface ComicInfoDao : RepositoryDao<UUID?, ComicInfo> {
    @Throws(SQLException::class)
    fun save(obj: ComicInfo)

    @Throws(SQLException::class)
    fun find(comic: String, linguagem: String): Optional<ComicInfo>

    @Throws(SQLException::class)
    fun find(id: UUID, comic: String, linguagem: String): Optional<ComicInfo>

    @Throws(SQLException::class)
    fun findEnvio(ultimo: LocalDateTime): List<ComicInfo>
}